src/test/scala/integration/TopicLoaderIntSpec.scala

```diff
@@ -247,6 +275,47 @@ class TopicLoaderIntSpec extends IntegrationSpecBase {
         }
       }
     }
+
+    "execute callback when finished loading and keep streaming per partition" in new TestContext {
+      val (preLoadPart1, postLoadPart1) = records(1 to 15).splitAt(10)
+      val (preLoadPart2, postLoadPart2) = records(16 to 30).splitAt(10)
+      val partitions: Long              = 2
+
+      withRunningKafka {
+        createCustomTopic(testTopic1, partitions = partitions.toInt)
+
+        publishToKafka(testTopic1, 0, preLoadPart1)
+        publishToKafka(testTopic1, 1, preLoadPart2)
+
+        val partitionedStream = TopicLoader
```

Comment:
does this test fail if this was to use just `loadAndRun`?

Reply:
Do you mean if we publish to one partition and then the other? It still passes with the `loadAndRun`:

```scala
    "execute callback when finished loading and keep streaming per partition" in new TestContext {
      val (preLoadPart1, postLoadPart1) = records(1 to 15).splitAt(10)
      val (preLoadPart2, postLoadPart2) = records(16 to 30).splitAt(10)
      val partitions: Long              = 2

      withRunningKafka {
        createCustomTopic(testTopic1, partitions = partitions.toInt)

        publishToKafka(testTopic1, 0, preLoadPart1)
        publishToKafka(testTopic1, 1, preLoadPart2)

        val ((callback, _), recordsProbe) =
          TopicLoader.loadAndRun[String, String](NonEmptyList.one(testTopic1)).toMat(TestSink.probe)(Keep.both).run()

        recordsProbe.request(
          preLoadPart1.size.toLong + postLoadPart1.size.toLong + preLoadPart2.size.toLong + postLoadPart2.size.toLong
        )
        recordsProbe
          .expectNextN(preLoadPart1.size.toLong + preLoadPart2.size.toLong)
          .map(recordToTuple) should contain theSameElementsAs preLoadPart1 ++ preLoadPart2

        whenReady(callback) { _ =>
          publishToKafka(testTopic1, 0, postLoadPart1)
          publishToKafka(testTopic1, 1, postLoadPart2)

          recordsProbe
            .expectNextN(postLoadPart1.size.toLong + postLoadPart2.size.toLong)
            .map(recordToTuple) should contain theSameElementsAs  postLoadPart1 ++ postLoadPart2
        }
      }
    }
```

Reply:
yeah so my point is i don't think we have a test that is proving this partitioned load functionality, considering the test passes even with `loadAndRun`